wallet/wallet.go

```diff
@@ -1615,17 +1615,21 @@ func (w *Wallet) NextAccount(name string) (uint32, error) {
 }
 
 // MasterPubKey returns the BIP0044 master public key for the passed account.
-//
-// TODO: This should not be returning the key as a string.
-func (w *Wallet) MasterPubKey(account uint32) (string, error) {
+func (w *Wallet) MasterPubKey(account uint32) (*hdkeychain.ExtendedKey, error) {
 	var masterPubKey string
 	err := walletdb.View(w.db, func(tx walletdb.ReadTx) error {
 		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
-		var err error
-		masterPubKey, err = w.Manager.GetMasterPubkey(addrmgrNs, account)
-		return err
+		var pKerr error
+		masterPubKey, pKerr = w.Manager.GetMasterPubkey(addrmgrNs, account)
+		return pKerr
 	})
-	return masterPubKey, err
+
+	if err != nil {
+		return nil, err
+	}
+
+	extKey, err := hdkeychain.NewKeyFromString(masterPubKey)
+	return extKey, err
```

Comment:
this can be done on a single line instead of two: `return hdkeychain.NewKeyFromString(masterPubKey)`